jerry-libc/include/assert.h

```diff
@@ -25,6 +25,11 @@ extern "C"
 #endif /* __cplusplus */
 
 #ifndef NDEBUG
+
+#ifndef __GNUC__
+#define __builtin_expect(expression, expected_value) (expression)
+#endif /* !__GNUC__ */
```

Comment:
I agree with @zherczeg, since the only direct use of `__builtin_expect()` is in `assert.h` this is best fixed by just replacing this invocation with `unlikely()`.